numbat/src/ast.rs

```diff
@@ -57,7 +57,11 @@ impl PrettyPrint for BinaryOperator {
 #[derive(Debug, Clone, PartialEq)]
 pub enum StringPart {
     Fixed(String),
-    Interpolation(Span, Box<Expression>),
+    Interpolation {
+        span: Span,
+        expr: Box<Expression>,
+        precision: Option<u64>,
```

Comment:
Instead of a `precision` field, we could have a `specifiers: Vec<Specifier>` field here.